### PR TITLE
Show single error string in Spice SQL REPL command line

### LINKED ISSUE #2018: SQL REPL error output should be improved
**Describe the bug**
We should not show internal errors for user errors or parsing errors.

i.e.

`SELECT * FROM non_existing_table` will produce an internal error when connected to a Spice.ai dataset:

`Status { code: Internal, message: "External error: Execution error: Unable to query Flight: Unable to query: status: InvalidArgument, message: \"Table 'tpch.orders' not found\\nstartLine 1\\nstartColumn 413\\nendLine 1\\nendColumn 423\\nSQL Query SELECT b.o_orderkey, b.o_custkey, b.o_orderstatus, b.o_totalprice, b.o_orderdate, b.o_orderpriority, b.o_clerk, b.o_shippriority, b.o_comment FROM (SELECT tpch.orders.o_orderkey, tpch.orders.o_custkey, tpch.orders.o_orderstatus, tpch.orders.o_totalprice, tpch.orders.o_orderdate, tpch.orders.o_orderpriority, tpch.orders.o_clerk, tpch.orders.o_shippriority, tpch.orders.o_comment FROM tpch.orders) AS b ORDER BY tpch.orders.o_orderkey ASC NULLS LAST LIMIT 1\", details: [], metadata: MetadataMap { headers: {\"date\": \"Fri, 19 Jul 2024 00:02:58 GMT\", \"content-type\": \"application/grpc\", \"strict-transport-security\": \"max-age=15724800; includeSubDomains\", \"alt-svc\": \"h3=\\\":443\\\"; ma=86400\", \"cf-ray\": \"8a567217ebe55a9c-MEL\", \"cf-cache-status\": \"DYNAMIC\", \"server\": \"cloudflare\"} }", source: None }`

**To Reproduce**

**Expected behavior**

**Spicepod**
Add the relevant `spicepod.yml` section.

**Output of the `describe table`**
Add any relevant `describe table` output here.

**Output of `explain query`**
Add any relevant `explain query` output here.

E.g. `explain select 1`
```
+---------------+--------------------------------------+
| plan_type     | plan                                 |
+---------------+--------------------------------------+
| logical_plan  | Projection: Int64(1)                 |
|               |   EmptyRelation                      |
| physical_plan | ProjectionExec: expr=[1 as Int64(1)] |
|               |   PlaceholderRowExec                 |
|               |                                      |
+---------------+--------------------------------------+
```

**spice, spiced, OS info**
- spice version: `spice version`
- spiced version: `spiced --version`
- OS info: `uname -a`

**Have you tried this on the latest `trunk` branch?**

**If possible, run `spiced` with DEBUG log level**

By setting the environment variable `SPICED_LOG="spiced=DEBUG,runtime=DEBUG,data_components=DEBUG,cache=DEBUG"`

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

Show user-friendly error in spice repl command line when spice query fails due to query planner failure.
* Show inline short error in command line
* Full error message can be retrieved using .error

Before:
```
sql> .error
```
` 
Status { code: Internal, message: "External error: Execution error: Unable to query Arrow Flight: Unable to query: status: InvalidArgument, message: \"Table 'tpch.lineitem' not found\\nstartLine 1\\nstartColumn 722\\nendLine 1\\nendColumn 734\\nSQL Query SELECT tpch.orders.o_orderpriority, count(1) AS order_count FROM tpch.orders WHERE (((tpch.orders.o_orderdate >= CAST('1993-07-01' AS TIMESTAMP)) AND (tpch.orders.o_orderdate < CAST((CAST('1993-07-01' AS DATE) + INTERVAL '3' MONTH) AS TIMESTAMP))) AND EXISTS (SELECT tpch.lineitem.l_orderkey, tpch.lineitem.l_partkey, tpch.lineitem.l_suppkey, tpch.lineitem.l_linenumber, tpch.lineitem.l_quantity, tpch.lineitem.l_extendedprice, tpch.lineitem.l_discount, tpch.lineitem.l_tax, tpch.lineitem.l_returnflag, tpch.lineitem.l_linestatus, tpch.lineitem.l_shipdate, tpch.lineitem.l_commitdate, tpch.lineitem.l_receiptdate, tpch.lineitem.l_shipinstruct, tpch.lineitem.l_shipmode, tpch.lineitem.l_comment FROM tpch.lineitem WHERE ((tpch.lineitem.l_orderkey = tpch.orders.o_orderkey) AND (tpch.lineitem.l_commitdate < tpch.lineitem.l_receiptdate)))) GROUP BY tpch.orders.o_orderpriority ORDER BY tpch.orders.o_orderpriority ASC NULLS LAST\", details: [], metadata: MetadataMap { headers: {\"date\": \"Tue, 30 Jul 2024 04:51:46 GMT\", \"content-type\": \"application/grpc\", \"strict-transport-security\": \"max-age=15724800; includeSubDomains\", \"alt-svc\": \"h3=\\\":443\\\"; ma=86400\", \"cf-ray\": \"8ab2bc48bad76826-SEA\", \"cf-cache-status\": \"DYNAMIC\", \"server\": \"cloudflare\"} }", source: None }
`

After:
image

## 🔨 Related Issues

- Close: https://github.com/spiceai/spiceai/issues/2018

### CODE REVIEW

File: crates/flight_client/src/lib.rs [hunk 1]
```diff
@@ -314,3 +316,15 @@ impl FlightClient {
         &self.username
     }
 }
+
+#[allow(clippy::needless_pass_by_value)]
+fn map_tonic_error_to_user_friendly_error(e: tonic::Status) -> Error {
+    if let Some(message) = e.message().split('\n').next() {
```
Comment (Jeadie) [inline crates/flight_client/src/lib.rs]: Is `\n` the common or best separator for these kind of errors? It works in the example in this PR, but I don't know how common/representative this is.

Comment (Sevenannn) [reply crates/flight_client/src/lib.rs]: Good point, I preserve the full message in flight client, and handle multi-line messages in repl by only showing 1 line of message in the command line

File: crates/flight_client/src/lib.rs [hunk 2]
```diff
@@ -314,3 +316,10 @@ impl FlightClient {
         &self.username
     }
 }
+
+#[allow(clippy::needless_pass_by_value)]
+fn map_tonic_error_to_message(e: tonic::Status) -> Error {
```
Comment (sgrebnov) [inline crates/flight_client/src/lib.rs:321]:
```suggestion
fn tonic_status_to_error(e: tonic::Status) -> Error {
```

File: crates/runtime/src/flight.rs [hunk 3]
```diff
@@ -214,7 +214,23 @@ impl Service {
                             flights.push(flight_batch.into());
                             Ok(flights)
                         }
-                        Err(e) => Err(Status::internal(e.to_string())),
+                        Err(e) => match e {
+                            DataFusionError::External(e) => {
+                                // Failed to downcast to datafusion execution error using the following method
+                                // if let Some(df_error) = e.downcast_ref::<DataFusionError>() {
+                                // }
+                                // Match Datafusion Error pattern with string
```
Comment (phillipleblanc) [inline crates/runtime/src/flight.rs]:
```suggestion
```

**Review by phillipleblanc — DISMISSED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**